gateway/path/resolver_test.go

```diff
@@ -0,0 +1,70 @@
+package path
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestResolvePath(t *testing.T) {
+	type args struct {
+		encodedPath string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    ResolvedPath
+		wantErr bool
+	}{
+		{
```

Comment:
Could you add cases that starts or ends with `/` ? I think the behaviour is inconsistent for these cases.